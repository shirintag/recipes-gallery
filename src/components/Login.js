import React from "react";

const Login = () => {
  return <div>HelloFresh</div>;
};

export default Login;
