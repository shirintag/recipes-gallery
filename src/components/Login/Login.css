.login-container {
  width: 100%;
  max-width: 400px;
  height: auto;
  margin: 100px auto;
  padding: 20px;
  align-content: center;
  text-align: center;
  box-sizing: border-box;
  border: 1px solid #e8e8e8;
  border-radius: 5px;
  background-color: white;
}

.login-input {
  width: 100%;
  height: 50px;
  margin: 0 auto 20px;
  padding: 15px;
  font-size: 12px;
  color: #9e9e9e;
  border: 1.2px solid #e8e8e8;
  border-radius: 5px;
  box-sizing: border-box;
}

::placeholder {
  color: #a5a5a5;
}

.login-button {
  display: inline-block;
  margin-top: 20px;
  padding: 15px 32px;
  border-radius: 2px;
  background-color: #5cb85c;
  font-size: 15px;
  letter-spacing: 0.99px;
  text-transform: uppercase;
  color: #ffffff;
  border: 1px solid #5cb85c;
  border-radius: 5px;
  text-align: center;
}
.login-button:hover {
  background-color: #449d44;
  border: 1px solid #449d44;
  color: white;
}
