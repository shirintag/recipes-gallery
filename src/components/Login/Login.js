import React, { useState } from "react";

import "./login.css";

const Login = (props) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  return (
    <div className="login-container">
      <h1 className="login-title f-title-login">Log in</h1>

      <form onSubmit={props.handleSubmit}>
        <input
          className="login-input"
          autoFocus
          value={email}
          type="email"
          placeholder="Email"
          onChange={(e) => setEmail(e.target.value)}
          required
        />
        <input
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          className="login-input"
          type="password"
          placeholder="Password"
          onChange={(e) => setPassword(e.target.value)}
          required
        />

        <input className="login-button" type="submit" value="login" />
      </form>
    </div>
  );
};

export default Login;
