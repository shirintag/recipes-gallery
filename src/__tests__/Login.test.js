import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Login from "components/Login/Login";

describe("Login", () => {
  it("should render", () => {
    const { container } = render(<Login />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call handle submit", () => {
    const handleSubmit = jest.fn();
    const { container } = render(<Login handleSubmit={handleSubmit} />);
    const form = container.querySelector("form");
    fireEvent.submit(form);
    expect(handleSubmit).toHaveBeenCalled();
  });
});
