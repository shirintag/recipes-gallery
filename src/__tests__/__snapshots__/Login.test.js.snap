// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render 1`] = `
<div
  class="login-container"
>
  <h1
    class="login-title f-title-login"
  >
    Log in
  </h1>
  <form>
    <input
      class="login-input"
      placeholder="Email"
      required=""
      type="email"
      value=""
    />
    <input
      class="login-input"
      placeholder="Password"
      required=""
      type="password"
      value=""
    />
    <input
      class="login-button"
      type="submit"
      value="login"
    />
  </form>
</div>
`;
